crates/core/tedge_api/src/entity_store.rs

```diff
@@ -144,6 +219,7 @@ impl EntityStore {
     pub fn update(
         &mut self,
         message: EntityRegistrationMessage,
+        external_id: EntityExternalId,
     ) -> Result<Vec<EntityTopicId>, Error> {
```

Comment:
Yeah, I was also initially thinking of accepting a closure to compute that external id when needed. But, since the external id computation relies on the state of the converter (the main device id), I thought it wouldn't work. That's why I decided to pass the computed value this way. And yes, it has the downside of unnecessarily computing the external id when an `@id` value is already present in the message. But, because of that closure limitation, I was forced to do it this way.